In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
df = pd.read_json('../Data2.json')
df.sort_index(inplace=True)

دانشگاه‌های پذیرفته شده: accUni

دانشگاه انتخاب شده:apUni

# accUni

In [3]:
df['accUni'].value_counts().count()

2505

** We have 2505 type of Universities in our data :)**

In [4]:
df['accUni']=df['accUni'].str.lower()
df['accUni'].value_counts().count()

2403

In [5]:
(df['accUni']=='n/a').value_counts()

False    2953
True      573
Name: accUni, dtype: int64

### 573 Missing Data in "accUni"

# apUni

In [6]:
df['apUni'].value_counts().count()

1607

In [7]:
df['apUni']=df['apUni'].str.lower()
df['apUni'].value_counts().count()

1376

In [8]:
df['apUni'].value_counts()

n/a                                                   426
politecnico di milano                                  46
kth                                                    46
university of alberta                                  32
concordia                                              32
utm                                                    27
ubc                                                    26
ntnu                                                   24
unsw                                                   23
concordia university                                   22
politecnico di torino                                  22
university of calgary                                  21
rwth aachen                                            20
mcgill                                                 19
university of toronto                                  19
chalmers                                               19
alberta                                                19
sfu           

### 426 Missing Data "apUni"

# bachelorsUni

In [9]:
df['bachelorsUni'].value_counts().count()

1367

In [10]:
df['bachelorsUni']=df['bachelorsUni'].str.lower()
df['bachelorsUni'].value_counts().count()

1221

In [11]:
df['bachelorsUni'].value_counts()

تهران                                            112
شریف                                              84
azad                                              82
university of tehran                              80
علم و صنعت                                        76
امیرکبیر                                          71
صنعتی اصفهان                                      61
amirkabir                                         46
تبریز                                             45
iust                                              45
sharif                                            44
شیراز                                             43
فردوسی مشهد                                       41
دانشگاه تهران                                     39
sharif university of technology                   38
خواجه نصیر                                        36
shiraz university                                 32
n/a                                               32
صنعتی شریف                                    

### 32 Missing Data in "bachelorsUni"

In [12]:
df['masterUni'].value_counts().count()

688

In [13]:
df['masterUni']=df['masterUni'].str.lower()
df['masterUni'].value_counts().count()

638

In [14]:
df['masterUni'].value_counts()

n/a                                                 1712
تهران                                                124
شریف                                                  77
university of tehran                                  61
امیرکبیر                                              57
علم و صنعت                                            43
صنعتی شریف                                            42
sharif                                                40
iust                                                  40
دانشگاه تهران                                         32
tehran                                                27
تربیت مدرس                                            25
شهید بهشتی                                            25
amirkabir                                             23
sharif university of technology                       21
خواجه نصیر                                            21
صنعتی امیرکبیر                                        21
صنعتی اصفهان                   

** 1712 Empty values. Does who want to apply for masters probably don't have MasterUni**

In [15]:
df[(df['apDegree']=='Doctorate') & (df['masterUni']=='n/a')]['masterUni'].value_counts()

n/a    167
Name: masterUni, dtype: int64

** Only 167 of people who want to apply for Doctorate don't have masterUni**

# rejUni

In [16]:
df['rejUni'].value_counts().count()

1259

In [17]:
df['rejUni']=df['rejUni'].str.lower()
df['rejUni'].replace([r'\-+','','none'],'n/a',regex=True,inplace=True)
df['rejUni'].value_counts().count()

1225

In [18]:
df['rejUni'].value_counts()

n/a                                                                                                                              2196
epfl                                                                                                                               10
ntnu                                                                                                                                8
cornell                                                                                                                             5
chalmers                                                                                                                            4
university of waterloo                                                                                                              4
university of helsinki                                                                                                              4
university of alberta                                         

** 2190 N/A Data for Rejected Universities**

In [19]:
df.to_json('Fix1.json',date_format='utf8')